src/Fixer/Alias/PowToExponentiationFixer.php

```diff
@@ -0,0 +1,157 @@
+            $this->fixPowToExponentiation(
+                $tokens,
+                $candidate[0], // functionNameIndex,
+                $candidate[1], // openParenthesisIndex,
+                $candidate[2], // closeParenthesisIndex,
+                $candidate[3]  // arguments
+            );
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Converts \'pow()\' to \'**\'.';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // must be run before BinaryOperatorSpacesFixer
+        return 1;
+    }
+
+    /**
+     * @param Tokens $tokens
+     *
+     * @return array[]
+     */
+    private function findPowCalls(Tokens $tokens)
+    {
+        $candidates = array();
+
+        // Minimal candidate to fix is seven tokens: pow(x,x);
+        $end = count($tokens) - 6;
+
+        // First possible location is after the open token: 1
+        for ($i = 1; $i < $end; ++$i) {
+            $candidate = $this->find('pow', $tokens, $i, $end);
+            if (null === $candidate) {
+                break;
+            }
+
+            $i = $candidate[1];
+
+            $arguments = $this->getArguments($tokens, $candidate[1], $candidate[2]);
+
+            if (2 !== count($arguments)) {
+                continue;
+            }
+
+            $candidate[] = $arguments;
+            $candidates[] = $candidate;
+        }
+
+        return $candidates;
+    }
+
+    /**
+     * @param Tokens         $tokens
+     * @param int            $functionNameIndex
+     * @param int            $openParenthesisIndex
+     * @param int            $closeParenthesisIndex
+     * @param array<int,int> $arguments
+     */
+    private function fixPowToExponentiation(Tokens $tokens, $functionNameIndex, $openParenthesisIndex, $closeParenthesisIndex, array $arguments)
+    {
+        // find the argument separator ',' directly after the last token of the first argument;
+        // replace it with T_POW '**'
+        $tokens->overrideAt(
+            $tokens->getNextTokenOfKind(reset($arguments), array(',')),
+            new Token(array(T_POW, '**'))
+        );
+
+        // clean up the function call tokens prt. I
+        $tokens[$closeParenthesisIndex]->clear();
+
+        // check if the arguments need to be wrapped in parenthesis
+        foreach (array_reverse($arguments, true) as $argumentStart => $argumentEnd) {
+            $needsParenthesis = false;
+            for ($i = $argumentStart; $i <= $argumentEnd; ++$i) {
+                if ($tokens[$i]->isWhitespace()
+                    || $tokens[$i]->isComment()
+                    || $tokens[$i]->isGivenKind(array(T_LNUMBER, T_VARIABLE, T_STRING, T_NS_SEPARATOR))
+                ) {
+                    continue;
+                }
+
+                if ($tokens[$i]->equals('(')) {
+                    $i = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $i);
```

Comment:
tokens between `[` and `]` could be skipped, too.
`pow($a[1 + 2], $b[3 + 4])` -> `$a[1 + 2] ** $b[3 + 4]`